pkgs/by-name/me/mealie/package.nix

```diff
@@ -0,0 +1,129 @@
+{ lib
+, callPackage
+, fetchFromGitHub
+, makeWrapper
+, nixosTests
+, python3
+, writeShellScript
+}:
+
+let
+  version = "1.1.0";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = "v${version}";
+    sha256 = "sha256-k2jMGJ6bdhkXootWl9cEdevRrGfggmHbkJZHZmhNLTM=";
+  };
+
+  frontend = callPackage (import ./mealie-frontend.nix src version) { };
+
+  python = python3.override {
+    packageOverrides = self: super: {
+      pydantic = self.pydantic_1;
+    };
+  };
+
+  # TODO  Add crfpp in runtime dependencies for ingredient quantities NLP
+
+in python.pkgs.buildPythonPackage rec {
+  pname = "mealie";
+  inherit version src;
+  pyproject = true;
+
+  patches = [
+    # See https://github.com/mealie-recipes/mealie/pull/3102
+    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
+    # So we can configure easily where the data is stored on the server
+    ./mealie_init_db.patch
+    ./mealie_logger.patch
+  ];
+
+  nativeBuildInputs = [
+    python.pkgs.poetry-core
+    python.pkgs.pythonRelaxDepsHook
+    makeWrapper
+  ];
+
+  dontWrapPythonPrograms = true;
+
+  pythonRelaxDeps = true;
+
+  propagatedBuildInputs = with python.pkgs; [
+    aiofiles
+    alembic
+    aniso8601
+    appdirs
+    apprise
+    bcrypt
+    extruct
+    fastapi
+    gunicorn
+    html2text
+    httpx
+    jinja2
+    lxml
+    orjson
+    passlib
+    pillow
+    psycopg2
+    pyhumps
+    pytesseract
+    python-dotenv
+    python-jose
+    python-ldap
+    python-multipart
+    python-slugify
+    pyyaml
+    rapidfuzz
+    recipe-scrapers
+    sqlalchemy
+    tzdata
+    uvicorn
+  ];
+
+  start_script = writeShellScript "start-mealie" ''
+    ${lib.getExe python.pkgs.gunicorn} "$@" -k uvicorn.workers.UvicornWorker mealie.app:app;
+  '';
+
+  postInstall = ''
+    mkdir -p $out/config $out/bin
+    rm -f $out/bin/*
+
+    substitute ${src}/alembic.ini $out/config/alembic.ini \
+      --replace-fail 'script_location = alembic' 'script_location = ${src}/alembic'
+
+    makeWrapper ${start_script} $out/bin/mealie \
+      --set PYTHONPATH "$out/${python.sitePackages}:${python.pkgs.makePythonPath propagatedBuildInputs}" \
+      --set STATIC_FILES ${frontend}
+  '';
+
+  checkInputs = with python.pkgs; [
+    pytestCheckHook
+  ];
+
+  passthru = {
+    init_db = pkg: writeShellScript "init-mealie-db" ''
+      export PYTHONPATH=${python.pkgs.makePythonPath ([pkg] ++ propagatedBuildInputs)}
+      ${python.interpreter} ${pkg}/${python.sitePackages}/mealie/db/init_db.py
+    '';
```

Comment:
Why is this a function? Make this a regular attr.

You can use `finalAttrs.finalPackage` to refer to the package itself from itself.

Reply:
Ok I missed that last piece of info thanks :sweat_smile:

Reply:
Oh indeed, it cannot do that. Sorry for the suggestion, I had assumed that it could. In that case you'd have to pass the package to itself or simply make this an executable in `libexec` as I initially recommended.

Reply:
I made a `init_db` script stored inside `$out/libexec/init_db`, I hope that's what you meant :smile: